src/bin/pgcopydb/env_utils.c

```diff
@@ -163,6 +166,88 @@ get_env_copy(const char *name, char *result, int maxLength)
 }
 
 
+/*
+ * process_env_line parses and processes a single line from .env file
+ */
+static bool
+process_env_line(void *ctx, char *line)
+{
+	EnvParserArray *parsers = (EnvParserArray *) ctx;
+	if (line[0] == '#')
+	{
+		return true;
+	}
+
+	/* split the line into key and value */
+	const char *key = line;
+	char *value = strchr(line, '=');
+	if (value == NULL)
+	{
+		return true;
+	}
+	*value = 0;
+	value++;
+	if (*value == '\n')
+	{
+		return true;
+	}
+
+	/* remove the newline character from the value */
+	value[strcspn(value, "\n ")] = 0;
+
+	for (int i = 0; i < parsers->count; i++)
+	{
+		if (strcmp(key, parsers->array[i].envname) == 0)
+		{
+			return get_env_value_using_parser(value, &parsers->array[i]);
+		}
+	}
+
+	return true;
+}
+
+
+/*
+ * get_env_using_parsers_from_file reads the environment variables from
+ * XDG_CONFIG_HOME/pgcopydb/.env file and uses the parsers to parse them.
+ */
+bool
+get_env_using_parsers_from_file(EnvParserArray *parsers)
+{
+	char envFilePath[BUFSIZE];
+	if (env_exists("XDG_CONFIG_HOME"))
+	{
+		char *configHome;
+		get_env_dup("XDG_CONFIG_HOME", &configHome);
+		sformat(envFilePath, sizeof(envFilePath), "%s/pgcopydb/.env", configHome);
+	}
+	else if (env_exists("HOME"))
+	{
+		char *homeDir;
+		get_env_dup("HOME", &homeDir);
+		sformat(envFilePath, sizeof(envFilePath), "%s/.config/pgcopydb/.env", homeDir);
+	}
```

Comment:
If we have a HOME environment variable defined, then we never look at te XDG_CONFIG_HOME path. I don't think that's the intended behavior?